plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2230,9 +2231,9 @@ private static boolean canUseInPredicate(ColumnMetadata column)
                 || type.equals(VARCHAR);
     }
 
-    private static Domain buildColumnDomain(ColumnMetadata column, DeltaLakeFileStatistics stats, List<String> canonicalPartitionColumns)
+    private static Domain buildColumnDomain(DeltaLakeColumnMetadata column, DeltaLakeFileStatistics stats, List<String> canonicalPartitionColumns)
     {
-        Optional<Long> nullCount = stats.getNullCount(column.getName());
+        Optional<Long> nullCount = stats.getNullCount(column.getPhysicalName().orElse(column.getName()));
```

Comment:
```
column.getPhysicalName().orElse(column.getName())
```

Is this done for backward compatibility?

Do you mind adding a comment for this behavior (I see the `nullCount` calculation performed also on `HiveMetastoreBackedDeltaLakeMetastore.java`) ?

I also see this pattern in the `XXXFileStatistics` files.

Should we create a properly commented method containing this logic to avoid code duplication?

Reply:
Now I think non-optional field might be better and always use them when accessing files. Please take another look and let me know your thoughts.